Applications/spire/include/spire/charting/cached_chart_model.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef SPIRE_CACHED_CHART_MODEL_HPP
+#define SPIRE_CACHED_CHART_MODEL_HPP
+#include "spire/charting/chart_model.hpp"
+
+namespace Spire {
+
+  //! Provides a caching mechanism for a ChartModel.
+  class CachedChartModel : public ChartModel {
+    public:
+
+      //! Signals an update to a candlestick.
+      using CandlestickSignal = Signal<void (const Candlestick& candle)>;
+
+      struct ChartRange {
+        ChartValue m_start;
+        ChartValue m_end;
+      };
+
+      //! Constructs a CachedChartModel that uses the supplied model.
+      /*!
+        \param model The model that supplies the data.
+      */
+      CachedChartModel(ChartModel& model);
+
+      ChartValue::Type get_x_axis_type() const override;
+
+      ChartValue::Type get_y_axis_type() const override;
+
+      QtPromise<std::vector<Candlestick>> load(ChartValue first,
+        ChartValue last) override;
+
+      //! Connects a slot to the candlestick signal.
```

Comment:
Remove documentation from overwritten methods unless the documentation adds new information not found in the base class.